### ISSUE TITLE
Issues decoding the ephemeral public key 

### ISSUE THREAD

**Issue by non-fungible-nelson** [hyperledger/besu#5177]
### Description
Incoming handshake, Besu throws the following exception on occasion:
`HandshakeException: Unable to create ECDH Key agreement due to Crypto engine failure`

### Versions (Add all that apply)
* Software version: Latest

**Comment by macfarla** [comment]
Could be related - #4601

**Comment by macfarla** [comment]
hypothesis - we are trying to handle a later message as a hello message - maybe something is happening out of order.

**Comment by macfarla** [comment]
analysing log file from AWS nodes with extra logging. at time of handshake we have the IP address, can see what client it belongs to if we connect to it later. 

Outgoing connections work ok, the problem is only with incoming connections

**Comment by pinges** [comment]
After running 5 Besu clients on mainnet for several days I have found one peer failing the inbound handshake and succeeding to do a handshake at another time. That peer was a fork of geth on the ubiq chain(https://github.com/ubiq/go-ubiq). I checked the handshake code on that fork for differences to geth, but could not find any.
Checking the IP addresses of clients that caused these failures have shown that a high percentage of these are located in China. This could point to software that uses the Ethereum discovery mechanism, but does not do the standard Ethereum handshake.
